rust/kernel/sync/guard.rs

```diff
@@ -64,11 +65,23 @@ impl<'a, L: Lock<M> + ?Sized, M> Guard<'a, L, M> {
     }
 }
 
+/// Specifies properties of a lock.
+pub trait LockInfo {
+    /// Determines if the data protected by a lock is writable.
+    type Writable: Bool;
```

Comment:
Would `const Writable: bool` work? Boolean const generics are supported on stable.

Reply:
It doesn't work, I actually talk a little bit about when I define `Bool`:

```rust
/// This is meant to be used in type states to allow booelan constraints in implementation blocks.
/// In the example below, the implementation containing `MyType::set_value` could _not_ be
/// constrained to type states containing `Writable = true` if `Writable` were a constant instead
/// of a type.
```

Here's a condensed exemple:

```rust
trait MyTypeState {
    const WRITABLE: bool;
}
struct MyType<T: MyTypeState>(std::marker::PhantomData<T>);
impl<T: MyTypeState<WRITABLE = true>> MyType<T> {}
```

This fails as follows:

```
error: cannot constrain an associated constant to a value
 --> <source>:5:21
  |
5 | impl<T: MyTypeState<WRITABLE = true>> MyType<T> {}
  |                     --------^^^----
  |                     |          |
  |                     |          ...cannot be constrained to this value
  |                     this associated constant...

error[E0220]: associated type `WRITABLE` not found for `MyTypeState`
 --> <source>:5:21
  |
5 | impl<T: MyTypeState<WRITABLE = true>> MyType<T> {}
  |                     ^^^^^^^^ associated type `WRITABLE` not found

error: aborting due to 2 previous errors
```

Using `Bool` works:
```rust
trait MyTypeState {
    type Writable: Bool;
}
struct MyType<T: MyTypeState>(std::marker::PhantomData<T>);
impl<T: MyTypeState<Writable = True>> MyType<T> {}
```

Reply:
You could do this, right?
```
struct MyType<const WRITABLE: bool>;
impl MyType<true> {}
```

Reply:
> You could do this, right?

Yes, I could, but that's not what I want.

How would I write two or more implementations where `WRITABLE` is `true`? (I have two for spinlocks, and will add a third that unconditionally disables/enables interrupts.)